File: Homework/hmwk_03_and7697/hmwk_03.java
// Duong, Andrew
// and7697
// 2019-06-17

import java.nio.file.Paths;
import java.nio.file.Files;
import java.util.Arrays;
import java.util.regex.*;

public class hmwk_03 {
  public static void processToken( String token )
  {
    // Replace the following line with your code to classify
    // the string in 'token' according to your three Regular
    // Expressions and print the appropriate message.

    if( Pattern.matches( "[aeiouAEIOU]+\\{\\)[0-9a-fA-F]+" , token ) )
      System.out.println( ">" + token + "< matches EffPea." );
    else if( Pattern.matches( "\\}[a-z0-9]*\\(" , token ))
      System.out.println( ">" + token + "< matches Stir." );
    else if( Pattern.matches( "\\@[0-9R-W]+\\#" , token ))
      System.out.println( ">" + token + "< matches Ent." );
    else
      System.out.println( ">" + token + "< does not match." );
  }

  public static void main( String[] args )
  {
    System.out.println( "processing tokens from " + args[ 0 ] + " ..." );

    try {
      Files.lines( Paths.get( args[ 0 ] ) ).filter(line -> line.length() > 0).forEachOrdered(
        line -> Arrays.stream( line.split( "\\s+" )  )
            .forEachOrdered( token -> processToken( token ) ) );
    } catch ( java.io.IOException e ) {
      e.printStackTrace();
    }
  }
}

File: Homework/hmwk_03_and7697/hmwk_03.py
# Duong, Andrew
# and7697
# 2019-06-18
#--------------------------------------------------
import re
import sys

#--------------------------------------------------
def processToken( token ) :
  # Replace the following line with your code to classify
  # the string in 'token' according to your three Regular
  # Expressions and print the appropriate message.

  if re.search( "^[aeiouAEIOU]+\\{\\)[0-9a-fA-F]+$" , token ) :
    print( ">" + token + "< matches EffPea." )
  elif re.search( "^\\}[a-z0-9]*\\($" , token ) :
    print( ">" + token + "< matches Stir." )
  elif re.search( "^\\@[0-9R-W]+\\#$" , token ) :
    print( ">" + token + "< matches Ent." )
  else :
    print( ">" + token + "< does not match." )

#--------------------------------------------------
def main() :
  fName = sys.argv[ 1 ]
  print( 'processing tokens from ' + fName + ' ...' )

  with open( fName, 'r' ) as fp :
    lines = fp.read().replace( '\r', '' ).split( '\n' )

  for line in lines :
    for token in line.split() :
      processToken( token )

#--------------------------------------------------
if ( __name__ == '__main__' ) :
  main()

#--------------------------------------------------
